genie-core/src/main/java/com/netflix/genie/core/jobs/JobLauncher.java

```diff
@@ -62,34 +64,38 @@ public JobLauncher(
         this.jobSubmitterService = jobSubmitterService;
         this.jobRequest = jobRequest;
         this.registry = registry;
+        this.submitTimer = this.registry.timer("genie.jobs.submit.timer");
     }
 
     /**
      * Starts the job setup and launch process once the thread is activated.
      */
     @Override
     public void run() {
-        try {
-            this.jobSubmitterService.submitJob(this.jobRequest);
-        } catch (final GenieException e) {
-            log.error("Unable to submit job due to exception: {}", e.getMessage(), e);
-            if (e instanceof GenieBadRequestException) {
-                this.registry.counter(MetricsConstants.GENIE_EXCEPTIONS_BAD_REQUEST_RATE).increment();
-            } else if (e instanceof GenieConflictException) {
-                this.registry.counter(MetricsConstants.GENIE_EXCEPTIONS_CONFLICT_RATE).increment();
-            } else if (e instanceof GenieNotFoundException) {
-                this.registry.counter(MetricsConstants.GENIE_EXCEPTIONS_NOT_FOUND_RATE).increment();
-            } else if (e instanceof GeniePreconditionException) {
-                this.registry.counter(MetricsConstants.GENIE_EXCEPTIONS_PRECONDITION_RATE).increment();
-            } else if (e instanceof GenieServerException) {
-                this.registry.counter(MetricsConstants.GENIE_EXCEPTIONS_SERVER_RATE).increment();
-            } else if (e instanceof GenieServerUnavailableException) {
-                this.registry.counter(MetricsConstants.GENIE_EXCEPTIONS_SERVER_UNAVAILABLE_RATE).increment();
-            } else if (e instanceof GenieTimeoutException) {
-                this.registry.counter(MetricsConstants.GENIE_EXCEPTIONS_TIMEOUT_RATE).increment();
-            } else {
-                this.registry.counter(MetricsConstants.GENIE_EXCEPTIONS_OTHER_RATE).increment();
+        this.submitTimer.record(() -> {
+                try {
+                    this.jobSubmitterService.submitJob(this.jobRequest);
+                } catch (final GenieException e) {
+                    log.error("Unable to submit job due to exception: {}", e.getMessage(), e);
+                    if (e instanceof GenieBadRequestException) {
```

Comment:
As in the String representation of the class name? That could be cleaner. We'll try to add it after I get through some more functional things.